Climate triggers: Turn on/off (Labs feature) (#45342)
home-assistant/home-assistant.github.io: source/_triggers/climate.turned_on.markdown

---
title: "Thermostat turned on"
trigger: climate.turned_on
domain: climate
description: "Triggers after one or more thermostats turn on, regardless of the mode."
related_triggers:
  - climate.turned_off
  - climate.hvac_mode_changed
---

The **Thermostat turned on** trigger fires after a thermostat {% term entity %} turns on, entering any operational mode (such as Heat, Cool, or Auto). The trigger doesn't care which specific mode the thermostat switches to—it only checks that it transitions from **Off** to any active mode. Use this trigger when you want to react as soon as the thermostat becomes active, regardless of whether it's heating, cooling, or in another mode.

> 

Author: I think that's just what those strings are called in the UI. 

core: https://github.com/home-assistant/core/blob/dev/homeassistant/components/climate/strings.json
<img width="589" height="251" alt="image" src="https://github.com/user-attachments/assets/48603c7a-d2a5-4c48-8af8-d5f3787c07d5" />

frontend: https://github.com/home-assistant/core/blob/dev/homeassistant/components/climate/strings.json

---
title: "Thermostat turned on"
trigger: climate.turned_on
domain: climate
description: "Triggers after one or more climate devices turn on, regardless of the mode."
related_triggers:
  - climate.turned_off
  - climate.hvac_mode_changed
---

> The **Thermostat turned on** trigger fires after a climate {% term entity %} turns on, entering any operational mode (such as Heat, Cool, or Auto). Climate entities include thermostats, air conditioners, heat pumps, and evaporative coolers. The trigger doesn't care which specific mode the device switches to—it only checks that it transitions from **Off** to any active mode. Use this trigger when you want to react as soon as the climate entity becomes active, regardless of whether it's heating, cooling, or in another mode.

Reviewer: ```suggestion
The **Thermostat turned on** trigger fires after a climate {% term entity %} turns on, entering any operational mode (such as Heat, Cool, or Auto). Climate entities include thermostats, air conditioners, heat pumps, and evaporative coolers. The trigger doesn't care which specific mode the device switches to. It only checks that it transitions from **Off** to any active mode. Use this trigger when you want to react as soon as the climate entity becomes active, regardless of whether it's heating, cooling, or in another mode.
```

> The **Thermostat turned on** trigger fires after a climate {% term entity %} turns on, entering any operational mode (such as Heat, Cool, or Auto). Climate entities include thermostats, air conditioners, heat pumps, and evaporative coolers. The trigger doesn't care which specific mode the device switches to—it only checks that it transitions from **Off** to any active mode. Use this trigger when you want to react as soon as the climate entity becomes active, regardless of whether it's heating, cooling, or in another mode.

Reviewer: Should these modes be in bold as well, for consistency with the **Off** in the same paragraph?
```suggestion
The **Thermostat turned on** trigger fires after a climate {% term entity %} turns on, entering any operational mode (such as **Heat**, **Cool**, or **Auto**). Climate entities include thermostats, air conditioners, heat pumps, and evaporative coolers. The trigger doesn't care which specific mode the device switches to—it only checks that it transitions from **Off** to any active mode. Use this trigger when you want to react as soon as the climate entity becomes active, regardless of whether it's heating, cooling, or in another mode.
```

> The **Thermostat turned on** trigger fires after a climate {% term entity %} turns on, entering any operational mode (such as Heat, Cool, or Auto). Climate entities include thermostats, air conditioners, heat pumps, and evaporative coolers. The trigger doesn't care which specific mode the device switches to. It only checks that it transitions from **Off** to any active mode. Use this trigger when you want to react as soon as the climate entity becomes active, regardless of whether it's heating, cooling, or in another mode.

Author: ```suggestion
The **Thermostat turned on** trigger fires after a climate {% term entity %} turns on, entering any operational mode (such as **Heat**, **Cool**, or **Auto**). Climate entities include thermostats, air conditioners, heat pumps, and evaporative coolers. The trigger doesn't care which specific mode the device switches to. It only checks that it transitions from **Off** to any active mode. Use this trigger when you want to react as soon as the climate entity becomes active, regardless of whether it's heating, cooling, or in another mode.
```

1. Go to {% my automations title="**Settings** > **Automations & scenes**" %}.
2. Open an existing automation, or select **Create automation** > **Create new automation**.
3. In the **When** section, select **Add trigger**.
4. Select what you want to monitor. Under **By target** (see [Targets](#targets)), pick the area your thermostat is in (like your bedroom or living room). You can also select a device, a specific entity, or a label.
5. From the triggers shown for that target, select **Thermostat turned on**.
6. Under **Trigger when** (see [Behavior](#behavior-with-multiple-targets)), pick **Each**, **First**, or **All** to control how the trigger behaves when multiple thermostats are targeted.
7. Under **For at least**, set how long the thermostat must stay on before the trigger fires. Leave it at zero to fire immediately.
8. Select **Save**.

### Options in the UI

{% options_ui %}
Trigger when:
  description: |
    When multiple thermostats are targeted, controls when the trigger fires:

    - **Each** (`any` in YAML, default): fire every time any targeted thermostat turns on.
    - **First** (`first` in YAML): fire only when the first of a group turns on.

>     - **All** (`last` in YAML): fire only after every targeted thermostat is on.

Reviewer: ```suggestion
    - **Each** (`any` in YAML, default): fires every time any targeted thermostat turns on.
    - **First** (`first` in YAML): fires only when the first of a group turns on.
    - **All** (`last` in YAML): fires only after every targeted thermostat is on.
```
